sbi/utils/mcmc_utils.py

```diff
@@ -0,0 +1,82 @@
+# This file is part of sbi, a toolkit for simulation-based inference. sbi is licensed
+# under the Apache License Version 2.0, see <https://www.apache.org/licenses/>
+
+import logging
+from typing import Callable, Optional, Union
+
+from numpy import ndarray
+from torch import Tensor
+
+from sbi.samplers.mcmc import SliceSamplerSerial, SliceSamplerVectorized
+from sbi.utils.torchutils import tensor2numpy
+
+
+def get_mcmc_samples(
+    potential_fn: Callable,
+    initial_params: Union[ndarray, Tensor],
+    num_samples: int = 1000,
+    vectorized: bool = True,
+    num_workers: int = 1,
+    num_chains: int = 1,
+    x: Optional[Union[ndarray, Tensor]] = None,
+    **kwargs,
+) -> ndarray:
+    """
+    Run slice MCMC sampling using a specified potential function.
+
+    Args:
+        potential_fn: User defined potential function. Must be of type Callable.
+        initial_params: Initial parameters for the MCMC sampler. First dimension
+            must match num_chains (default=1).
+        num_samples: Number of samples to draw.
+        vectorized: Whether to use the vectorized implementation.
+        num_workers: Number of workers to use for vectorized sampling.
+        x: The input to the potential function.
+
+    Returns:
+        Samples from the posterior distribution of shape
+            [num_chains, num_samples, parameter_dim]
+    """
+    # Check that the first dimension of initial_params matches num_chains.
+    assert initial_params.shape[0] == num_chains, (
+        "first dimension of initial_params must match num_chains"
+    )
+
+    # Convert initial_params to NumPy array if necessary.
+    if isinstance(initial_params, Tensor):
+        initial_params = tensor2numpy(initial_params)
+
+    if x is None:
+        logging.warning(
+            "x has not been provided. Using unconditional potential function."
+        )
+    else:
+        if isinstance(x, Tensor):
+            x = tensor2numpy(x)
+
```

Comment:
I think it would still make sense to use the `mcmc_transform` in this function, as it would be good to transform `theta` into an unconstrained space for the sampler to work well. This can be constructed in this function if the user provides the prior.